src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -157,23 +168,33 @@ private static string GenerateUniqueName(IParameterSymbol parameter, ImmutableAr
             return uniqueName;
         }
 
-        private IPropertySymbol CreateProperty(
-            IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts)
+        private async Task<IPropertySymbol> CreatePropertyAsync(
+            Document document, IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts, CancellationToken cancellationToken)
         {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var requireAccessibilityModifiers = options.GetOption(CodeStyleOptions.RequireAccessibilityModifiers);
+
             foreach (var rule in rules)
             {
                 if (rule.SymbolSpecification.AppliesTo(SymbolKind.Property, Accessibility.Public))
                 {
                     var uniqueName = GenerateUniqueName(parameter, parameterNameParts, rule);
 
+                    // We do not add an accessibility level if the user turned the setting off
+                    var accessibilityLevel = Accessibility.Public;
+                    if (requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.Never || requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.OmitIfDefault)
+                    {
+                        accessibilityLevel = Accessibility.NotApplicable;
+                    }
```

Comment:
This is wrong.  This is not what this option controls.  The option controls whether or modifiers are required in the case where they're redundant.  But 'public' is never redundant.  If you leave it off, the property will be private.  And that's not what is being generated.

Reply:
Oh I see, thank you for the clarification! I'll fix that ASAP. The option should still apply for fields though, correct?

Reply:
Most of the time.  But there is a case for VB where it does not apply.  Thanks!